cli/tests/test_duplicate_command.rs

```diff
@@ -2427,6 +2427,96 @@ fn test_rebase_duplicates() {
     ");
 }
 
+#[test]
+fn test_duplicate_description_template() {
+    let test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let work_dir = test_env.work_dir("repo");
+
+    create_commit(&work_dir, "a", &[]);
+    create_commit(&work_dir, "b", &["a"]);
+    create_commit(&work_dir, "c", &["b"]);
+
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&work_dir), @r"
+    @  7e4fbf4f2759   c
+    ○  1394f625cbbd   b
+    ○  2443ea76b0b1   a
+    ◆  000000000000
+    [EOF]
+    ");
+
+    // Test duplicate_commits()
+    test_env.add_config(r#"templates.duplicate_description = "concat(description, '\n(cherry-picked from commit ', commit_id, ')')""#);
+    let output = work_dir.run_jj(["duplicate", "a"]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Duplicated 2443ea76b0b1 as yostqsxw b8771261 a
+    [EOF]
+    ");
+    insta::assert_snapshot!(get_log_output(&work_dir), @r"
+    @  7e4fbf4f2759   c
+    ○  1394f625cbbd   b
+    ○  2443ea76b0b1   a
+    │ ○  b87712611d0f   a
+    ├─╯
+    ◆  000000000000
+    [EOF]
+    ");
+    let output = work_dir.run_jj(["show", "--summary", "b8"]);
+    insta::assert_snapshot!(output, @r"
+    Commit ID: b87712611d0fea7693cddb93b18ab1e1542b530f
+    Change ID: yostqsxwqrltovqlrlzszywzslusmuup
+    Author   : Test User <test.user@example.com> (2001-02-03 08:05:08)
+    Committer: Test User <test.user@example.com> (2001-02-03 08:05:15)
+
+        a
+
+        (cherry-picked from commit 2443ea76b0b1c531326908326aab7020abab8e6c)
+
+    A a
+    [EOF]
+    ");
+
+    // Test duplicate_commits_onto_parents()
+    let output = work_dir.run_jj(["duplicate", "7e", "-B", "b8"]);
```

Comment:
nit: better to use bookmark name? you'll have to update hashes if something added to tests.

Reply:
Would it really be useful to use the bookmark names for some of the commits, but not others? (Only the ones from `create_commit` have the bookmarks set, and not the ones created by `jj duplicate`.

(I also think I ran into some issues when using the bookmark names for `create_commit` commits, but I can try again if you think it's worth it.)

Reply:
Sure, but we would still have to update the `jj show` commands for the newly duplicated commits. Do you think there's a better way to refer to those too?

Reply:
Maybe dump all commits at the end (by `jj log -T..`)?

btw, when we have to use hashes, I would suggest using 8-12 characters. The shortest hash is harder to spot, and has higher chance that it points to different commit by accident.